public/scripts/controller.js

```diff
@@ -0,0 +1,157 @@
+/*jshint esversion: 6 */
+var gameApp = angular.module('routerApp.mainController',['ngAnimate', 'ui.bootstrap','pusher-angular']);
+
+
+
+gameApp.controller('mainController', ['$scope','$pusher', '$http','$rootScope', '$window', '$filter', '$state',
+    function ($scope,$pusher, $http, $rootScope, $window, $filter, $state) {
+
+        $rootScope.userName = $window.localStorage.user;
+        $rootScope.logged = $window.localStorage.logged;
+
+    if (typeof $rootScope.subscribed == 'undefined'){
+        $rootScope.client = new Pusher('1404da432444b8f9e7aa', {
+                cluster: 'eu',
+                encrypted: true
+            });
+        $rootScope.pusher = $pusher($rootScope.client);
+
+            var channel = $rootScope.pusher.subscribe('messages');
+        $rootScope.subscribed = true;
+
+        channel.bind('my_event',
+            function() {
+                $http.get('/phrases')
+                    .success(function(data) {
+                        data.forEach(function(item){
+                            if(item.last_user != $window.localStorage.user){
+                                item.editable = true;
+                            }else{
+                                item.editable = false;
+                            }
+                        });
+                        $scope.phrases = data;
+                    });
+            }
+        );
+
+        }
+
+       $http.get('/phrases')
+            .success(function(data) {
+               data.forEach(function(item){
+                   if(item.last_user != $window.localStorage.user){
+                        item.editable = true;
+                   }else{
+                       item.editable = false;
+                   }
+               });
+                $scope.phrases = data;
+
+            });
+
+        $scope.addWord = function(id) {
+            var postData = {'word': $scope.word, 'user': $window.localStorage.user};
+            $http.post('api/phrases/' + id + '/words', postData);
+
+        };
+
+        $scope.getHistory = function (id) {
+            $rootScope.hitories = [];
+            var text = '';
+            var phrase = $filter('filter')($scope.phrases, { id: id})[0];
+            phrase.words.forEach(function (word) {
+                var time = '(' + word.time.slice(0, -6) + ')';
```

Comment:
and what if time format will change? 
Feel free to ignore this comment but I would use moment.js